src/plugins/vis_augmenter/public/actions/external_action_action.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ActionByType, createAction, ExternalActionContext } from '../../../ui_actions/public';
+import { isPointInTimeAnnotation } from '../vega';
+
+export type ExternalActionActionContext = ExternalActionContext;
+export const ACTION_EXTERNAL_ACTION = 'ACTION_EXTERNAL_ACTION';
+
+declare module '../../../ui_actions/public' {
+  export interface ActionContextMapping {
+    [ACTION_EXTERNAL_ACTION]: ExternalActionActionContext;
+  }
+}
+
+export function createExternalActionAction(): ActionByType<typeof ACTION_EXTERNAL_ACTION> {
+  return createAction<typeof ACTION_EXTERNAL_ACTION>({
+    type: ACTION_EXTERNAL_ACTION,
+    id: ACTION_EXTERNAL_ACTION,
+    shouldAutoExecute: async () => true,
+    execute: async (context: ExternalActionActionContext) => {
+      if (isPointInTimeAnnotation(context.data.item)) {
+        if (context.data.event === 'click') {
+          // TODO: show events flyout
+        } else if (context.data.event === 'mouseover') {
+          // TODO: show custom tooltip
+        }
+      }
+    },
```

Comment:
Should we have these actions split into 2? One for triggering the flyout opening, and another for triggering the custom tooltip? Curious of thoughts from @ashwin-pc on this as well.

Reply:
Depends on whether we want to keep events from the visualization generic and have the handler distinguish based off of the data. This way we will keep the events minimal and avoid adding specific info into the general framework.

Reply:
Yeah - I don't have a strong preference either way. Agreed we don't want to overload too specific of events here. Maybe one per `VisLayer` implementation makes sense - in this particular case, handling click/mouseover/mouseout actions for the `PointInTimeEventsVisLayer`.

Reply:
On second thought, perhaps we do want to have these split up. We will need a mechanism for disabling the click event behavior, but still enabling the custom tooltip behavior in the view events flyout. Do you see a way we could pass that logic here? Or if we split into 2 events maybe (1 for custom click event, 1 for custom hover event), be able to disable one and enable the other?

Reply:
@amsiglan thanks. This makes sense to me in the context of the click event; I can defined a UIAction that listens for this trigger to open the view events flyout. But there's 2 things I'd like to clarify:
1. For the scenario of disabling the clicking, but still enabling the custom tooltips, how do you propose I do this? One solution I could see is altering `addVisEventSignalsToSpecConfig()` to only have `on` for the mouse events, and just omit the click event.
2. This is maybe a bit of a miss on my part when reviewing the latest revisions made, but for the tooltips, how should we get sufficient context passed such that a UIAction can render the tooltip correctly? The original proposal was making changes to the implemented [TooltipHandler](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/feature/feature-anywhere/src/plugins/vis_type_vega/public/vega_view/vega_tooltip.js#L58) class, which is using the out-of-the-box `tooltip()` function provided by vega to have a customized tooltip to render it OUI-style. It requires a lot of extra context, such as the container the vega view is in, the view itself, and other positioning parameters.

Reply:
For 1. We don't need to disable the clicks, we could just check if the flyout is already open then just no-op in the event listener.
For 2. we will need to do a quick POC and check if the provided event has all the information we need. I see it has the view and the position related information
<img width="622" alt="image" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/114732919/a6d6a849-7613-468b-8c67-5710a411181f">

Reply:
> We don't need to disable the clicks, we could just check if the flyout is already open then just no-op in the event listener.
How would you propose this? We don't persist global flyout state so how would we be able to make such a check within the UIAction?

Reply:
@ohltyler Why do we need to disable click? what is the interaction pattern that we want to achieve with that?

Reply:
So user can't infinitely re-open the flyout when clicking - details in the issue https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3317